src/storage/src/source/postgres.rs

```diff
@@ -172,6 +177,24 @@ impl SourceRender for PostgresSourceConnection {
                 .insert(idx, output);
         }
 
+        // Both the `replication` and the `statistics` operator want to observe the `resume_uppers`
+        // stream, so spawn a task to "fork" it.
+        let (resume_uppers_tx_1, resume_uppers_rx_1) = mpsc::unbounded_channel();
+        let (resume_uppers_tx_2, resume_uppers_rx_2) = mpsc::unbounded_channel();
+        let resume_uppers_1 = UnboundedReceiverStream::new(resume_uppers_rx_1);
+        let resume_uppers_2 = UnboundedReceiverStream::new(resume_uppers_rx_2);
+        mz_ore::task::spawn(|| format!("resume-uppers-fork:{}", config.id), async move {
+            let mut resume_uppers = pin!(resume_uppers);
+            while let Some(upper) = resume_uppers.next().await {
+                let r1 = resume_uppers_tx_1.send(upper.clone());
+                let r2 = resume_uppers_tx_2.send(upper);
+
+                if r1.is_err() && r2.is_err() {
+                    break;  // both receivers disconnected
+                }
+            }
+        });
```

Comment:
This part is a major smell, but I don't know how to avoid it. The `statistics` operator needs to look at the committed uppers to write them into its statistics updates. The `replication` operator needs to look at them to send keepalives back to the PG instance.

It seems like the stream we get here originally comes from a `watch`, so we could consider passing in the `watch` instead of an opaque `impl Stream`. But that'd probably require touching a whole bunch of unrelated code.